[RDY] Fix return status coverity issues

nwneisen opened the pull request:
Fix coverity issues related to unchecked return values.

justinmk commented:
Thanks. Lint errors can be checked like this:

    make lint

Also let's use ELOG (not EMSG) for all cases, or maybe WLOG.

nwneisen commented:
I used ELOG or WLOG depending on how severe the FAIL appeared to be.  Should additional error handling be added for the ELOG situations?

justinmk commented:
Restarted build, tests were broken by a python package issue which has been fixed.

> Should additional error handling be added for the ELOG situations?

I'd say no, given that Vim was ignoring the retvals in these cases, it's probably just noise.

nwneisen committed:
coverity/13969: Check return status of u_save

Check the return status after saving the cursor position.  Add a
log for a FAIL status.

nwneisen committed:
coverity/13713: Check u_save_cursor return status

Check the status after saving cursor line.  Add error message
for FAIL status.

nwneisen committed:
coverity/13700: Check return status of win_split

Check the return status after splitting a window.  Add a log for
a FAIL status.

nwneisen committed:
coverity/13702: Check ml_open return status

Check the return status after opening a new memline.  Add a log
for a FAIL status.

nwneisen committed:
coverity/13709: Check return status of fseek

Check the return status after removing a duplicate word.  Add a
log for a nonzero return status.

justinmk commented:
For these cases it's really necessary to look at the implementations of the failing calls to make a decision.

justinmk commented on src/nvim/spell.c:
```diff
@@ -2487,7 +2487,9 @@ buf_T *open_spellbuf(void)
   buf->b_p_swf = true;        // may create a swap file
-  ml_open(buf);
+  if (ml_open(buf) == FAIL) {
+    ELOG("Error opening a new memline");
```
`ml_open` already does `IEMSG` if it fails, so we don't need to log or do anything, only return.

But I think the nature of open_spellbuf is that it never fails, which is why the error wasn't checked here. If we return early then we would return nonsense.

So this should be changed to ignore the return value, i.e.

    (void)ml_open(buf);

justinmk commented on src/nvim/tag.c:
```diff
@@ -2407,8 +2407,10 @@ jumpto_tag (
-        postponed_split_flags);
+    if (win_split(postponed_split > 0 ? postponed_split : 0,
+                  postponed_split_flags) == FAIL) {
+      WLOG(_("Error splitting window"));
```
Inspecting the implementation of win_split() shows that it does `EMSG` for all failure cases, so we don't need to log it here.

Easiest thing to do here is ignore the return value (otherwise we need to do some cleanup and might require some refactoring.  jumpto_tag() can deal with a failed split, so it's no big deal.

    (void)win_split(...);

justinmk commented on src/nvim/spellfile.c:
```diff
@@ -5378,7 +5378,9 @@ spell_add_word (
           }
-          fseek(fd, fpos_next, SEEK_SET);
+          if (fseek(fd, fpos_next, SEEK_SET) <= 0) {
+            ELOG("Error seeking to next position");
```
If fseek() fails we should `break`, just like the case where `mch_fopen` fails a few lines above.

justinmk commented:
merged after making the above changes.

nwneisen commented:
Thanks @justinmk.  I appreciate you taking the time to explain these to me.
